### PR TITLE
[REF] switch from (undeclared) class property to local variable.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor code cleanup

Before
----------------------------------------
undeclared class property used, extraneous config call

After
----------------------------------------
local var used, extraneous call removed

Technical Details
----------------------------------------
We either needed to declare the variable on the class & make it a local variable. Switching
to a local var reflects the fact it is never accessed from outside thie function & improves readability.

grepping for _userOptions returns nothing after this.

I also removed an extraneous config singleton call. I can't see a strong case that either of
these changes will affect the intermittent fails but ... maybe?

Comments
----------------------------------------
@seamuslee001 @mattwire either of you OK to check this - should be pretty simple as it's just preliminary to my next step (using the api to get the contribution rows rather than Selector object - in order to simplify some performance fixes & hopefully deal to the intermittent test fail)

### CODE REVIEW

File: CRM/Contact/Page/View/UserDashBoard.php [hunk 1]
```diff
@@ -58,16 +58,12 @@ class CRM_Contact_Page_View_UserDashBoard extends CRM_Core_Page {
   public function __construct() {
     parent::__construct();
 
-    $check = CRM_Core_Permission::check('access Contact Dashboard');
-
-    if (!$check) {
+    if (!CRM_Core_Permission::check('access Contact Dashboard')) {
```
Comment (mattwire) [inline CRM/Contact/Page/View/UserDashBoard.php]: ok

File: CRM/Contact/Page/View/UserDashBoard.php [hunk 2]
```diff
@@ -58,16 +58,12 @@ class CRM_Contact_Page_View_UserDashBoard extends CRM_Core_Page {
...
       CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/dashboard', 'reset=1'));
     }
 
     $this->_contactId = CRM_Utils_Request::retrieve('id', 'Positive', $this);
-
-    $session = CRM_Core_Session::singleton();
-    $userID = $session->get('userID');
+    $userID = CRM_Core_Session::singleton()->getLoggedInContactID();
```
Comment (mattwire) [inline CRM/Contact/Page/View/UserDashBoard.php]: ok

File: CRM/Contact/Page/View/UserDashBoard.php [hunk 3]
```diff
@@ -186,9 +181,9 @@ public function buildUserDashBoard() {
     $this->assign('dashboardElements', $dashboardElements);
 
     // return true when 'Invoices / Credit Notes' checkbox is checked
-    $this->assign('invoices', $this->_userOptions['Invoices / Credit Notes']);
+    $this->assign('invoices', $dashboardOptions['Invoices / Credit Notes']);
 
-    if (!empty($this->_userOptions['Groups'])) {
+    if (!empty($dashboardOptions['Groups'])) {
```
Comment (mattwire) [inline CRM/Contact/Page/View/UserDashBoard.php]: This variable is only used within this function. So ok with this if it helps set the scene for further refactoring

**Review by mattwire — APPROVED** [review]
I've reviewed the code and given this a brief test - ok to merge

**Comment by eileenmcnaughton** [comment]
thanks @mattwire

**Pull request merged**